meta.go

```diff
@@ -29,51 +35,123 @@ var (
 	date    = "unknown"
 )
 
-var mkdirAll = os.MkdirAll
-var stat = os.Stat
-var writeFile = ioutil.WriteFile
-var readFile = ioutil.ReadFile
-var fprintf = fmt.Fprintf
-
 var metaKeyValidator = regexp.MustCompile(`^(\w+(-*\w+)*)+(((\[\]|\[(0|[1-9]\d*)\]))?(\.(\w+(-*\w+)*)+)*)*$`)
 var rightBracketRegExp = regexp.MustCompile(`\[(.*?)\]`)
 
-// getMeta prints meta value from file based on key
-func getMeta(key string, metaSpace string, metaFile string, output io.Writer, jsonValue bool, lastSuccessfulMetaRequest *fetch.LastSuccessfulMetaRequest) error {
-	metaFilePath := metaSpace + "/" + metaFile + ".json"
+// MetaSpec encapsulates the parameters usually from CLI so they are more readable and shareable than positional params.
+type MetaSpec struct {
+	MetaSpace                    string
+	SkipFetchNonexistentExternal bool
+	MetaFile                     string
+	JsonValue                    bool
```

Comment:
consider adding a comment regarding what this 'JsonValue' is or should be used for or what it contains.

Reply:
also `golint` recommends: `meta.go:46:2: struct field JsonValue should be JSONValue`

Reply:
addressed all golint warnings